### DISCUSSION TITLE
Combine non-virtual DOM with virtual DOM

### DISCUSSION THREAD

**Discussion by Bit0r** [vuejs/core-vapor#290] [Ideas]
Is there a way to combine virtual DOM and native DOM operations? Compile relatively simple and "static" reactive operations into native DOM operations, while continuing to use virtual DOM for some more complex and "dynamic" operations.

This is a way of "hydration", where part of the reactions are compiled into native DOM operations by vapor at compile time to improve performance, while the more complex parts are handled by the runtime virtual DOM.

This allows for the benefits of both.

**Comment by LittleSound** [comment]
Combine Vue Classical (virtual DOM) and Vue Vapor (non-virtual DOM), it has been our plan all along.

You can see this link: https://blog.vuejs.org/posts/2022-year-in-review#vapor-mode
